Added support for the individual release lookup feature of the forge api.

ericparton opened the pull request:
More information about this part of the forge api can be found here:
https://forgeapi.puppetlabs.com/#!/release/getRelease

This will allow the Puppetlabs forge ruby client and anything that uses it (such as r10k) to interact with Pulp puppet repositories.

mhrivnak commented:
ok test

ericparton commented:
Hello, I just made one minor change to a regex so that it would allow an additional character. I also saw that the check against the earlier commit was failing due to "No test results found." Is there anything I need to do about this?

Thanks

dkliban commented:
ok test

bmbouter commented:
@ericparton Thanks for submitting something that looks really usable. I don't have too much familiarity with using this, and it's likely other users won't either. This PR needs some docs and a release note since it's great new functionality. This would be a new thing, not a bugfix right?

Guidelines for writing docs where Pulp is integrating with other things (ie: r10k)... Normally we don't document specific commands since those can change but a claim of what the functionality is and requirements on a user to get that functionality working.

I'll review the code now too and leave more comments

bmbouter commented:
Please squash these two commits and force push a new version. Also please add this into your commit message to relate your commit to the pulp.plan.io issue.

```
https://pulp.plan.io/issues/1536
closes #1536
```

bmbouter commented on pulp_puppet_plugins/pulp_puppet/forge/views/releases.py:
```diff
@@ -93,20 +96,61 @@ def _get_credentials(headers):
+        match = POST_36_URL_PATTERN.match(path)
+        if match:
+            path_parameter = match.group(1)
+            match = MODULE_PATH_PARAM_PATTERN.match(path_parameter)
```
Consider not reusing `match` but instead keep them as two separate filenames. It makes the `if` logic confusing.

bmbouter commented:
@ericparton I'm done w/ my review and this code looks good. The only thing it really needs before merging are some docs. Once those are pushed let me know and I can re-review/merge.

ericparton commented:
@bmbouter Yeah, I would consider this a new thing instead of a bugfix. I've gone ahead and addressed your earlier comment about the variable name, squashed the commits, and added some documentation and a release note. There was some similar documentation in the "Recipes" section for the puppet command line tools, so I added another section there for r10k.

bmbouter commented:
@ericparton Thanks a lot for the docs. I read through them and they look good enough to me.

@dkliban Can you ensure that this PR passes w/ the full pulp-smash suite?

dkliban commented:
@ericparton I will test this PR and if the tests pass, I will merge it.

dkliban commented:
ok test

ericparton commented:
@dkliban Great, thanks!

dkliban commented:
@ericparton I have not forgotten about this PR. I've been trying to resolve some other issues with the plugin first.

ericparton commented:
@dkliban No worries. Thanks for the heads up.

bmbouter commented:
In my up-to-date dev environment I verified that this PR causes 0 pulp-smash test failures.

```
[vagrant@dev ~]$ psmash
~ ~
flake8 . --ignore D203
pylint -j 4 --reports=n --ignore-imports=y --disable=I \
        docs/conf.py \
        setup.py \
        tests \
        pulp_smash/__init__.py \
        pulp_smash/__main__.py \
        pulp_smash/api.py \
        pulp_smash/cli.py \
        pulp_smash/config.py \
        pulp_smash/constants.py \
        pulp_smash/exceptions.py \
        pulp_smash/selectors.py \
        pulp_smash/utils.py
No config file found, using default configuration
pylint -j 4 --reports=n --disable=I,duplicate-code pulp_smash/tests/
No config file found, using default configuration
python -m unittest discover --start-directory tests --top-level-directory .
.........................................................
----------------------------------------------------------------------
Ran 57 tests in 0.111s

OK
..........................s...........s.......s....ssss...........ssss.....................................................................................s............
----------------------------------------------------------------------
Ran 164 tests in 551.398s

OK (skipped=12)
```

Based on that I'm merging it.
